### ISSUE TITLE
[web] updateTheme property on react-dom

### ISSUE THREAD

**Issue by hedikasmanto** [react-native-training/react-native-elements#1747]
**Explain what you did**
I use "react-native-web" + "react-native-elements" with ThemeProvider like this below render

```
<ThemeProvider>
        <View>
          <Header
            leftComponent={{ icon: 'menu', color: '#fff' }}
            centerComponent={{ text: 'MY TITLE', style: { color: '#fff' } }}
            rightComponent={{ icon: 'home', color: '#fff' }}
          />
     </View>
</ThemeProvider>

```
**Expected behaviour**
No Error should be thrown

**Describe the bug**
Warning: React does not recognize the `updateTheme` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `updatetheme` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in div (created by Text)
    in Text (at Text.js:12)
    in TextElement (at withTheme.js:42)
    in Themed.Text (at renderNode.js:18)
    in div (created by View)
    in View (at Header.js:26)
    in Children (at Header.js:94)
    in div (created by View)
    in View (created by ImageBackground)
    in ImageBackground (at Header.js:70)
    in Header (at withTheme.js:42)
    in Themed.Header (at App.js:21)
    in div (created by View)
    in View (at App.js:20)
    in ThemeProvider (at App.js:19)
    in App (at js/index.js:6)

**Your Environment (please complete the following information):**

| software              | version |
| --------------------- | ------- |
| react-native-elements |  1.0.0       |
| react-native          |   web 0.10      |
| npm or yarn           |    yarn 1.13     |

**Comment by xavier-villelegier** [comment]
@hedikasmanto Does it work even with this warning ?
Actually the `ThemeProvider` injects two props: `theme` and `updateTheme`, that provides you the ability to update the theme dynamically.

I guess react-native-web transforms the `View` into a `div` or something similar, so you end up with a `div` that has an `updateTheme` props, obviously not a standard DOM props.

Maybe you could just remove the `View` that is wrapping your `Header` component ? If you really want to get rid of this warning, and you doesn't need to update your theme dynamically, you can still write a custom component and exclude the `updateTheme` props manually.

**Comment by hedikasmanto** [comment]
Yes. It still working. And I found, withTheme HoC cause this issue..

**Comment by xavier-villelegier** [comment]
Then you should make sure that the decorated component is a custom component, to avoid passing these two `theme` and `updateTheme` props to an any HTML component (`div`, `span`…)

**Comment by hedikasmanto** [comment]
Thanks for your information. I have created PR for this. Please review
https://github.com/react-native-training/react-native-elements/pull/1758

**Comment by xavier-villelegier** [comment]
Closing this for now as it doesn't affect the functionality. However, if someone knows how avoid the warning without removing completely the `updateTheme` props, please let us know here 🙏

**Comment by bdombro** [comment]
Here is a [patch-package](http://npmjs.org/package/patch-package) patch which seemingly resolves the issue without losing functionality. It works by simply not passing unused props to react-native components. So far the patch works for Text, Input, and Searchbar. There may be other RNE elements that need patching, which I'll hopefully have time to test against in the near future.


```
diff --git a/node_modules/react-native-elements/src/input/Input.js b/node_modules/react-native-elements/src/input/Input.js
index e727fad..b28b153 100644
--- a/node_modules/react-native-elements/src/input/Input.js
+++ b/node_modules/react-native-elements/src/input/Input.js
@@ -76,6 +76,8 @@ class Input extends React.Component {
       labelStyle,
       labelProps,
       theme,
+      updateTheme,
+      replaceTheme,
       ...attributes
     } = this.props;

diff --git a/node_modules/react-native-elements/src/searchbar/SearchBar-default.js b/node_modules/react-native-elements/src/searchbar/SearchBar-default.js
index 39b7ebc..3712342 100644
--- a/node_modules/react-native-elements/src/searchbar/SearchBar-default.js
+++ b/node_modules/react-native-elements/src/searchbar/SearchBar-default.js
@@ -75,6 +75,7 @@ class SearchBar extends React.Component {
       showLoading,
       loadingProps,
       placeholderTextColor = theme.colors.grey3,
+      onClear,
       ...attributes
     } = rest;

diff --git a/node_modules/react-native-elements/src/text/Text.js b/node_modules/react-native-elements/src/text/Text.js
index b1b26c2..c738e34 100644
--- a/node_modules/react-native-elements/src/text/Text.js
+++ b/node_modules/react-native-elements/src/text/Text.js
@@ -17,6 +17,9 @@ const TextElement = props => {
     h2Style,
     h3Style,
     h4Style,
+    theme,
+    updateTheme,
+    replaceTheme,
     ...rest
   } = props;
```

**Comment by OlegBrazhnichenko** [comment]
@hedikasmanto, @xavier-villelegier, 
I found how to get rid of that error . 
One of my header components looks like: 
```
<div theme="[object Object]" text="Signals" dir="auto" class="css-text-901oao css-textOneLine-bfa6kz r-fontSize-adyw6z r-fontWeight-vw2c0b" style="color: rgb(0, 178, 223);">
    Signals
</div>
```
And I noticed that `theme="[object Object]"` looks really strange. Also I found out that only one of three my components have that prop. 

My header looks like: 
```
<Header
    containerStyle={{...}}
   
    leftComponent={
        <View style={{paddingLeft: 20}}>
            <Icon
                name="three-bars"
                size={30}
                color="#00AEE1"
                onPress={()=>updateMenuState(!isMenuShown)}
            />
        </View>
    }
    centerComponent={{
        text: I18n.t(`routes.${navigation.state.routeName}`),
        style: styles.headerTitle,
    }}
    rightComponent={
        <View style={{paddingRight: 15}}>
                <Icon
                    name="kebab-vertical"
                    size={30}
                    color="#00AEE1"
                    onPress={() => {}}
                />
        </View>
    }
/> 
```
Only center component is not custom and have theme prop. So I tried to remove it with one more custom component and error is gone. 

Final code : 
```
<Header
    containerStyle={{...}}
   
    leftComponent={
        ...
    }
    centerComponent={
       <Text style={styles.headerTitle}>
           {I18n.t(`routes.${navigation.state.routeName}`)}
       </Text>
      }
    rightComponent={
        ...
    }
/> 
```
Also I don't use any ThemeProvider or withTheme HOC, so this solution might not help you if you use them. 

Hope it help you :)

**Comment by iRoachie** [comment]
Fixing this in the next patch release

**Comment by iRoachie** [comment]
Fix released in [`v1.2.4`](https://github.com/react-native-training/react-native-elements/releases/tag/v1.2.4)
